openc3/lib/openc3/operators/microservice_operator.rb

```diff
@@ -64,27 +64,65 @@ def update
       # Detect new and changed microservices
       @new_microservices = {}
       @changed_microservices = {}
+      @removed_microservices = {}
       @microservices.each do |microservice_name, microservice_config|
+        parent = microservice_config['parent']
         if @previous_microservices[microservice_name]
+          previous_parent = @previous_microservices[microservice_name]['parent']
           if @previous_microservices[microservice_name] != microservice_config
+            # CHANGED
             scope = microservice_name.split("__")[0]
             Logger.info("Changed microservice detected: #{microservice_name}\nWas: #{@previous_microservices[microservice_name]}\nIs: #{microservice_config}", scope: scope)
-            @changed_microservices[microservice_name] = microservice_config
+            if parent or previous_parent
+              if parent == previous_parent
+                # Same Parent - Respawn parent
+                @changed_microservices[parent] = @microservices[parent] if @microservices[parent] and @previous_microservices[parent]
+              elsif parent and previous_parent
+                # Parent changed - Respawn both parents
+                @changed_microservices[parent] = @microservices[parent] if @microservices[parent] and @previous_microservices[parent]
+                @changed_microservices[previous_parent] = @microservices[previous_parent] if @microservices[previous_parent] and @previous_microservices[previous_parent]
+              elsif parent
+                # Moved under a parent - Respawn parent and kill standalone
+                @changed_microservices[parent] = @microservices[parent] if @microservices[parent] and @previous_microservices[parent]
+                @removed_microservices[microservice_name] = microservice_config
+              else # previous_parent
+                # Moved to standalone - Respawn previous parent and make new
+                @changed_microservices[previous_parent] = @microservices[previous_parent] if @microservices[previous_parent] and @previous_microservices[previous_parent]
+                @new_microservices[microservice_name] = microservice_config
+              end
+            else
+              # Respawn regular microservice
+              @changed_microservices[microservice_name] = microservice_config
+            end
           end
```

Comment:
What's going on with all the re-parenting logic? How would this play out in a real system?

Reply:
Part of this change is the ability to break out microservices from the multi_microservice.   This allows for scaling and isolation if necessary.  There is no way currently to switch between two parents, but there are ways to go from having a parent, to not, and from not, to having a parent.